src/pages/index.js

```diff
@@ -14,81 +16,190 @@ const addCardAddButton = document.querySelector(selectors.cardAddButton);
+avatarPopup.setEventListeners();
 
-// ! ||--------------------------------------------------------------------------------||
-// ! ||                                 Event Listeners                                ||
-// ! ||--------------------------------------------------------------------------------||
+// Event Listeners
 addCardAddButton.addEventListener("click", () => {
   newCardPopup.open();
   cardFormValidator.toggleButtonState();
 });
 
 updateProfileButton.addEventListener("click", () => {
   const userData = userInfo.getUserInfo();
-  profileHeadingInput.value = userData.userName;
-  profileDescriptionInput.value = userData.userDescription;
+  profileHeadingInput.value = userData.name;
+  profileDescriptionInput.value = userData.about;
   editProfileFormValidator.resetValidation();
   updateProfilePopup.open();
 });
 
-// ! ||--------------------------------------------------------------------------------||
-// ! ||                                 Event Handlers                                 ||
-// ! ||--------------------------------------------------------------------------------||
-function handleAddCardFormSubmit(data) {
-  const card = renderCard(data);
-  cardSection.addItem(card);
+updateAvatarButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+// Functions
+function handleValidation(form) {
+  form.enableValidation();
+}
+
+function handleAddCardFormSubmit(cardData) {
+  newCardPopup.showButtonProgress(true);
+  api.addNewCard(cardData.name, cardData.link)
+    .then((res) => {
+      const card = renderCard(res);
+      cardSection.addItem(card);
+      newCardPopup.reset();
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      newCardPopup.showButtonProgress(false);
+    });
 }
 
 function handleImageClick(name, link) {
   newImagePopup.open(name, link);
 }
 
 function handleProfileSubmit(userData) {
-  userInfo.setUserInfo(userData);
-}
\ No newline at end of file
+  updateProfilePopup.showButtonProgress(true);
+  api.updateUserInfo(userData.name, userData.about)
+    .then((user) => {
+      userInfo.setUserInfo(user);
+      updateProfilePopup.reset();
+      updateProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      updateProfilePopup.showButtonProgress(false);
+    });
+}
+
+function handleDeleteCard(cardData) {
+  deletePopup.open();
+  deletePopup.setSubmitAction(() => {
+    deletePopup.showButtonProgress(true);
+    api.deleteCard(cardData._id)
+      .then(() => {
+        cardData.handleRemoveCard();
+        deletePopup.close();
+      })
+      .catch((err) => {
+        console.error(`Error: ${err}`);
+      })
+      .finally(() => {
+        deletePopup.showButtonProgress(false);
+      });
+  });
+}
+
+function handleLikeIcon(cardData) {
+  api.setLike(cardData._id, cardData._isLiked)
```

Comment:
`_id`
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods if you want to use them `outside` a class because the `low dash` shows a **private** method which is used only inside the class.

Reply:
* You haven't corrected https://github.com/Marcus3278/se_project_aroundtheus-main/pull/5#discussion_r1612383020